alephium/contracts/token_bridge/wrapped_alph.ral

```diff
@@ -0,0 +1,34 @@
+Contract WrappedAlph(
+    mut totalWrapped: U256
+) {
+    @using(preapprovedAssets=true, assetsInContract=true, permissionCheck=false)
+    pub fn wrapAndTransfer(
+        fromAddress: Address,
+        totalAmount: U256
+    ) -> () {
+        totalWrapped = totalWrapped + totalAmount
```

Comment:
Do we want to check the caller of this method?

Reply:
It would be great if we can check the caller, but I don't have a good way to do this 😓

Reply:
Good point. Perhaps we can check the `callerCodeHash`? I guess we know the code hash of the `WrappedAlphPool` when we create `WrappedAlph` contract.

Since we only have one copy of `WrappedAlphPool`, maybe that is enough? Otherwise it feels a bit dangerous since the attacker can use an arbitrary contract id to initiate the `WrappedAlphPool` here.

Reply:
Hmm, maybe even the above is not enough...

Do we need to actually have a field in `WrappedAlph` to remember the contract id of the `WrappedAlphPool` after `WrappedAlphPool` is created, and only allow these two methods to be called when the contract id is set?

Reply:
Ok, maybe it is ok, since we restricted the amount of tokens that can be received here

```
WrappedAlphPool(callerContractId!()).receiveWrappedAlph{
   self -> selfTokenId!(): totalAmount
}(self, totalAmount)
```

Reply:
I'm thinking that maybe we can merge `WrappedAlph` to `WrappedAlphPool` if the `WrapperAlph` is only used for the bridge. Because we only have one `WrappedAlph` and one `WrappedAlphPool` instance.

Reply:
Yeah, I thought about that too, I kinda like the fact that `WrapperAlph` makes everything conceptually the same: a "token". But perhaps it is a good idea to merge it :thinking: